viewing_party/party.py

```diff
@@ -1,23 +1,155 @@
 # ------------- WAVE 1 --------------------
 
+
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    return {"title": title, "genre": genre, "rating": rating}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    new_watchlist = []
+
+    for i in range(len(watchlist)):
+        if title == watchlist[i]["title"]:
+            add_to_watched(user_data, watchlist[i])
+        else:
+            new_watchlist.append(watchlist[i])
+    user_data["watchlist"] = new_watchlist
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    if not user_data["watched"]:
+        return 0
+
+    ratings_sum = sum([movie["rating"] for movie in user_data["watched"]])
+    movie_count = len(user_data["watched"])
+
+    return ratings_sum/movie_count
+
+
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+
+    genre_count = {}
+
+    for movie in user_data["watched"]:
+        if movie["genre"] in genre_count:
+            genre_count[movie["genre"]] += 1
+        else:
+            genre_count[movie["genre"]] = 1
+
+    return max(genre_count, key=genre_count.get)
+
+
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):
+    friends_watched = []
+    list_of_movies = []
+
+    for i in user_data["friends"]:
+        for j in i["watched"]:
+            friends_watched.append(j["title"])
+
+    for i in user_data["watched"]:
+        if i["title"] not in set(friends_watched):
+            list_of_movies.append(i)
+
+    return list_of_movies
+
+
+def get_friends_unique_watched(user_data):
+    user_titles = []
+    for movie in user_data["watched"]:

```

Comment:
This appears to repeat some logic found in the previous function, so you could definitely break it out of both functions and move it into a helper that could be leveraged in both! If you get the sense that you're duplicating logic, it's a good indicator that you can probably do some refactoring!